netx.go

```diff
@@ -5,24 +5,78 @@ package netx
 import (
 	"context"
 	"net"
+	"strings"
 	"sync/atomic"
 	"time"
 )
 
 var (
-	dial           atomic.Value
-	dialUDP        atomic.Value
-	listenUDP      atomic.Value
-	resolveTCPAddr atomic.Value
-	resolveUDPAddr atomic.Value
-
+	dial               atomic.Value
+	dialUDP            atomic.Value
+	listenUDP          atomic.Value
+	resolveTCPAddr     atomic.Value
+	resolveUDPAddr     atomic.Value
+	NAT64Prefix        atomic.Value
 	defaultDialTimeout = 1 * time.Minute
 )
 
 func init() {
 	Reset()
 }
 
+type NAT64PrefixHolder struct {
+	expiration time.Time
+	prefix     []byte
+}
+
+func ResetNAT64Prefix() {
```

Comment:
Who would use this and when?

Reply:
the idea is if we detect a network change or something elsewhere, invalidate -- I thought we had something external to this watching for network changes but perhaps not.

Reply:
Makes some sense. In such a case, is there a predictable error that would come back when using the old NAT64 prefix and if so, could we just detect that in here and reset internally?